src/DependencyGraph/ResolutionRequest.js

```diff
@@ -167,24 +171,21 @@ class ResolutionRequest {
 
           response.setResolvedDependencyPairs(mod, filteredPairs);
 
-          filteredPairs.forEach(([depName, modDep]) => {
-            p = p.then(() => {
-              if (!visited[modDep.hash()]) {
+          return Promise.all(
+            filteredPairs
+              .filter(([, modDep]) => !visited[modDep.hash()])
+              .map(([depName, modDep]) => {
                 visited[modDep.hash()] = true;
-                response.pushDependency(modDep);
-                if (recursive) {
-                  return collect(modDep);
-                }
-              }
-              return null;
-            });
-          });
-
-          return p;
+                return Promise.all([modDep, recursive ? collect(modDep) : []]);
```

Comment:
Not super easy because I actually had to disable this for www because of how slow it is :(

You could potentially try it on react-native after I land my diffs but landcastle is giving me trouble.

I used lower numbers, I think 16, 24 or 32 should be the max for concurrent promises that access the file system here.